### ISSUE TITLE
Is ZAP beneficial also for Leaky Relu

### ISSUE THREAD

**Issue by MohitLamba94** [gilshm/zap#2]
Very interesting paper.
I see the point in predicting zero activations which reduces the MAC. And central to this argument is the widespread usage of ReLU.

Will ZAP be beneficial if my CNN is using Leaky Relu and thus i expect not many activations will be zero?

Thankyou

**Comment by gilshm** [comment]
Hi, thanks for the feedback.

Obviously, with Leaky ReLU the number of zero-valued activations will decrease in favor of negative values, as you said.
I wonder how large the negative values turn out to be with Leaky ReLUs. Also, I wonder how important the negative values to the loss function and what will happen if some of them will be rounded to zero during inference.

Since I don't have much hands-on experience with Leaky ReLUs, I don't have these answers; however, I do believe that ZAP can be modified to support this version of ReLU to some extent.

**Comment by MohitLamba94** [comment]
Yes even I believe the idea presented in paper is interesting and by working more on it a generalized framework can be designed which generalizes ZAP to various activations and not just stop at ReLU. This is important because many image generations models such as denoising, low-light enhancement have actively spoken against usage of ReLU and have adopted Leaky ReLU with negative slope of 0.2.
